core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -212,6 +212,8 @@ class HadoopRDD[K, V](
       logInfo("Input split: " + split.inputSplit)
       val jobConf = getJobConf()
 
+      // TODO: there is a lot of duplicate code between this and NewHadoopRDD and SqlNewHadoopRDD
```

Comment:
Yep, `SQLNewHadoopRDD` started as a copy of `NewHadoopRDD` and then diverged some more. It would be great to try to clean this up at some point.